chips/stm32f446re/src/usart.rs

```diff
@@ -477,36 +498,33 @@ impl hil::uart::Uart<'a> for Usart<'a> {}
 
 impl dma1::StreamClient for Usart<'a> {
     fn transfer_done(&self, pid: dma1::Dma1Peripheral) {
-        match pid {
-            dma1::Dma1Peripheral::USART2_TX => {
-                self.usart_tx_state.set(USARTStateTX::Transfer_Completing);
-                self.enable_transmit_complete_interrupt();
-            }
-            dma1::Dma1Peripheral::USART2_RX => {
-                // In case of RX, we can call the client directly without having
-                // to trigger an interrupt.
-                if self.usart_rx_state.get() == USARTStateRX::DMA_Receiving {
-                    self.disable_rx();
-                    self.usart_rx_state.set(USARTStateRX::Idle);
-
-                    // get buffer
-                    let buffer = self.rx_dma.map_or(None, |rx_dma| rx_dma.return_buffer());
-
-                    let length = self.rx_len.get();
-                    self.rx_len.set(0);
-
-                    // alert client
-                    self.rx_client.map(|client| {
-                        buffer.map(|buf| {
-                            client.received_buffer(
-                                buf,
-                                length,
-                                ReturnCode::SUCCESS,
-                                hil::uart::Error::None,
-                            );
-                        });
+        if pid == self.tx_dma_pid {
+            self.usart_tx_state.set(USARTStateTX::Transfer_Completing);
+            self.enable_transmit_complete_interrupt();
+        } else if pid == self.rx_dma_pid {
+            // In case of RX, we can call the client directly without having
+            // to trigger an interrupt.
+            if self.usart_rx_state.get() == USARTStateRX::DMA_Receiving {
+                self.disable_rx();
+                self.usart_rx_state.set(USARTStateRX::Idle);
+
+                // get buffer
+                let buffer = self.rx_dma.map_or(None, |rx_dma| rx_dma.return_buffer());
+
+                let length = self.rx_len.get();
+                self.rx_len.set(0);
+
+                // alert client
+                self.rx_client.map(|client| {
+                    buffer.map(|buf| {
+                        client.received_buffer(
+                            buf,
+                            length,
+                            ReturnCode::SUCCESS,
+                            hil::uart::Error::None,
+                        );
                     });
```

Comment:
@garasubo Thank you for implementing this. After thinking bit more about the macro issue, I wanted to implement exactly the same thing, so we could avoid introducing the macro. :-) I had planned on refactoring `transfer_done` and test it out the change week before suggesting potential change to #1263, but looks like you got to it sooner. Nice!